### ISSUE TITLE
Enable `monofont` key in YAML for powerpoint_presentation output format

### ISSUE THREAD

**Issue by apreshill** [rstudio/rmarkdown#2081]
The pandoc documentation provides `monofont` as an option for this output format:

https://pandoc.org/MANUAL.html#variables-for-powerpoint

But this option is not defined in the `rmarkdown::powerpoint_presentation` output format. 
https://rmarkdown.rstudio.com/docs/reference/powerpoint_presentation.html

When I try to use it in my YAML:

```
output: 
  powerpoint_presentation:
    monofont: 'DM Mono' # doesn't work
```

I get this error:

```
Error in (function (toc = FALSE, toc_depth = 2, number_sections = FALSE,  : 
  unused argument (monofont = "DM Mono")
Calls: <Anonymous> -> create_output_format -> do.call
Execution halted
```

**Comment by cderv** [comment]
Any pandoc variable can be set in the yaml frontmatter. 

So you can do 
```yaml
output: powerpoint_presentation
monofont: 'DM Mono'
```

And it should work. 

This is how to change font for PDF output already for example. 

Some docs with example mainly with PDF 
- https://bookdown.org/yihui/rmarkdown-cookbook/latex-variables.html
- https://bookdown.org/yihui/rmarkdown/pdf-document.html
- https://bookdown.org/yihui/rmarkdown/template-pandoc.html

Maybe we should also give example with several formats. 

These pandoc variables are often shared accross format and we can rely on Pandoc itself to deal with them. Nothing is done in R through the output format, that is why you don't set them inside the format function. 

Does it help ?

**Comment by apreshill** [comment]
No, doesn't help- I tried it there first and it knits without errors, but also does not seem to have any effect. I can never change the monofont in my output- it is always "Courier" no matter what I try,

**Comment by cderv** [comment]
Then this is an issue to look into. The doc you linked to in pandoc says it is a pandoc variable. It should work if set that way in the YAML as other pandoc variables. 

However there could be something with the name of the font - is it available to PowerPoint with this name or another syntaxe ? (Other case, no space, ... )

Thanks for opening the issue !

**Comment by apreshill** [comment]
I've tried to set it to 10 different fonts available in my system now, and I'm unable to hack the magical incantation. I found this:
https://github.com/jgm/pandoc/issues/5491#issuecomment-490632381
to suggest usage.

And this:
https://learnbyexample.github.io/customizing-pandoc/

But that is about as far as I can figure out 😞

**Comment by cderv** [comment]
I am wondering if there is not something wrong with Pandoc considering these two examples with this file
* `test.md`
````markdown
---
title: "toy_example"
monofont: Source Code Pro
---

# Slide 1

With `2+2` inline

# Slide 2

```r
1+1
```
````

1. Works: `pandoc -t pptx -o test.pptx -V"monofont=Source Code Pro" test.md `
2. Does not work: `pandoc -t pptx -o test.pptx test.md `

I believe the second should also work and the Pandoc variable is passed in YAML header. 

I will maybe asked this on Pandoc discuss.

**Comment by apreshill** [comment]
Thank you for investigating! I'm happy to close this and open an issue in pandoc repo if that is where it belongs.

**Comment by cderv** [comment]
It seems like a Pandoc issue. I opened one https://github.com/jgm/pandoc/issues/7187 following a question in Pandoc discussion https://groups.google.com/g/pandoc-discuss/c/ZhHpDusB_y0

We'll see.

**Comment by cderv** [comment]
@apreshill issue is now fixed upstream in Pandoc (https://github.com/jgm/pandoc/commit/e22d1fbb144d63d595c5db9225dd912b09fd938f) so it should be available in next Pandoc version. This will lead to a requirement on Pandoc version for this to work. 

So maybe we can have an exception and add this variable as an argument in `pptx_presentation` so that we set the value at the command line ? 

For now, with your Pandoc version, you need to that: 
```yaml
output: 
  powerpoint_presentation:
    pandoc_args: ["--variable", "monofont=DM Mono"]
```

We could make this happen
```yaml
output: 
  powerpoint_presentation:
    monofont: 'DM Mono'
```
It would be an exception compared to the way it works currently. 
But something similar has been asked already in https://github.com/rstudio/bookdown/issues/918 i.e Pandoc variable specific to a format should be in the output format option.

**Comment by apreshill** [comment]
I need to check these additional arguments:

https://github.com/jgm/pandoc/pull/5573

- monofont
- code-font
- code-font-size
- code-margin

These options for PPT appear to not be documented?
